### PR TITLE
Add GitHub Actions + JooqMock 1.1.1

### PR DESCRIPTION
**Pull request by jackblanc**
A merge of backend-scaffold's master branch.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -41,26 +40,14 @@
             <target>1.8</target>
           </configuration>
         </plugin>
-
-        <!-- Used for running the main class -->
```
Comment (connernilsen) [inline pom.xml:45]: oh wait we ended up having to add this back in I think

Comment (connernilsen) [reply pom.xml:45]: NEVER MIND I'M BLIND

File: service/src/test/java/com/codeforcommunity/processor/ProcessorImplTest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
+package com.codeforcommunity.processor;
```
Comment (connernilsen) [inline service/src/test/java/com/codeforcommunity/processor/ProcessorImplTest.java:1]: This file can be deleted, was just being used for `@BeforeEach` testing

**Comment by connernilsen** [comment]
I think the `@BeforeEach` annotation needs to be added to `setup()` in `ReservationProcessorImplTest` and calls to `setup()` can be removed now

**Comment by jackblanc** [comment]
@connernilsen can that be ticketed separately?

**Comment by connernilsen** [comment]
Sure why not

**Pull request merged**